### PR TITLE
tests: Drop unstable_worker and tests based on it

### PR DESCRIPTION
**Pull request by kalikiana**
Going by the observation that none of us is really happy with how these tests work we might be better off with other existing tests covering the same thing, or even new tests.

See: https://progress.opensuse.org/issues/80274

### CODE REVIEW

File: t/05-scheduler-full.t [hunk 1]
```diff
@@ -136,146 +136,6 @@ subtest 'Scheduler worker job allocation' => sub {
...
-    @workers = broken_worker(@$worker_settings, 3, 'out of order');
-    wait_for_worker($schema, 5);
-    $allocated = scheduler_step();
-    is @$allocated, 0, 'scheduler does not consider broken worker for allocating job';
-    stop_workers;
-    dead_workers($schema);
-
-    # simulate a worker in idle state that rejects all jobs assigned to it
-    @workers = rejective_worker(@$worker_settings, 3, 'rejection reason');
-    wait_for_worker($schema, 5);
```
Comment (Martchus) [inline t/05-scheduler-full.t:162]: The tests with the broken and rejective worker should stay. These workers should behave consistently.

File: t/05-scheduler-full.t [hunk 2]
```diff
@@ -136,146 +136,6 @@ subtest 'Scheduler worker job allocation' => sub {
...
-        last if $jobs->find(99982)->state eq OpenQA::Jobs::Constants::DONE;
-        sleep .2;
-    }
-
-    my $job = $jobs->find(99982);
-    is $job->state, OpenQA::Jobs::Constants::DONE,
-      'running job set to done if its worker re-connects claiming not to work on it anymore';
-    is $job->result, OpenQA::Jobs::Constants::INCOMPLETE,
-      'running job incompleted if its worker re-connects claiming not to work on it anymore';
-    like $job->reason, qr/abandoned: associated worker .+:\d+ re-connected but abandoned the job/, 'reason is set';
```
Comment (Martchus) [inline t/05-scheduler-full.t:224]: Not sure whether this is already covered (and also actually checked!) somewhere else.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: t/05-scheduler-full.t (hunk 2 above)
Comment (kalikiana) [reply t/05-scheduler-full.t:224]: I was hoping to see coverage results telling me that.

**Comment by okurz** [comment]
Please also crosscheck to keep code coverage up. The idea was that we delete test code assuming there is coverage provided by other tests. Better start small, so as @Martchus suggested, keep the "broken and rejective workers"

**Comment by okurz** [comment]
@kalikiana maybe it helps readability of coverage reports if we merge https://github.com/os-autoinst/openQA/pull/3728 first, wait for tests in master to produce proper coverage base reports and then you simply try again and we see what parts are not covered after your code removal. The idea was really to not need to add sophisticated tests but rather remove unreliabe existing stuff, right?

**Comment by okurz** [comment]
please rebase now so that we can see how the current coverage report looks like

**Pull request closed without merging**